homeassistant/components/vesync/light.py

```diff
@@ -0,0 +1,87 @@
+"""Support for VeSync dimmers."""
+import logging
+import math
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    SUPPORT_BRIGHTNESS,
+    LightEntity,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .common import VeSyncDevice
+from .const import DOMAIN, VS_DISCOVERY, VS_DISPATCHERS, VS_LIGHTS
+
+_LOGGER = logging.getLogger(__name__)
+
+DEV_TYPE_TO_HA = {
+    "ESD16": "light",
+    "ESWD16": "light",
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up lights."""
+
+    async def async_discover(devices):
+        """Add new devices to platform."""
+        _async_setup_entities(devices, async_add_entities)
+
+    disp = async_dispatcher_connect(
+        hass, VS_DISCOVERY.format(VS_LIGHTS), async_discover
+    )
+    hass.data[DOMAIN][VS_DISPATCHERS].append(disp)
+
+    _async_setup_entities(hass.data[DOMAIN][VS_LIGHTS], async_add_entities)
+    return True
+
+
+@callback
+def _async_setup_entities(devices, async_add_entities):
+    """Check if device is online and add entity."""
+    dev_list = []
+    for dev in devices:
+        if DEV_TYPE_TO_HA.get(dev.device_type) == "light":
+            dev_list.append(VeSyncDimmerHA(dev))
+        else:
+            _LOGGER.warning(
+                "%s - Unknown device type - %s", dev.device_name, dev.device_type
+            )
+            continue
+
+    async_add_entities(dev_list, update_before_add=True)
+
+
+class VeSyncDimmerHA(VeSyncDevice, LightEntity):
+    """Representation of a VeSync dimmer."""
+
+    def __init__(self, dimmer):
+        """Initialize the VeSync dimmer device."""
+        super().__init__(dimmer)
+        self.dimmer = dimmer
+
+    def turn_on(self, **kwargs):
+        """Turn the device on."""
+        if ATTR_BRIGHTNESS in kwargs:
+            brightness = int(kwargs[ATTR_BRIGHTNESS])
+            if brightness < 1:
+                brightness = 1
+            elif brightness > 255:
+                brightness = 100
+            else:
+                brightness = math.ceil((brightness / 255) * 100)
```

Comment:
Can we reduce some of the precision loss with `round` at the end instead of `math.ceil` ?

Reply:
Clamp brightness instead of the if block:

`brightness = max(1, min(brightness, 100))`

Reply:
Simplified it a bit. `ceil` is needed here to avoid rounding to 0 which is invalid input to the vesync api.

Reply:
`brightness = max(1, min(brightness, 100))`

The clamp should take care of that since the min will be 1

Reply:
You're right -- I swapped the conversion from 1-255 to 1-100 and the clamp and swapped to using round.